lib/operators/general-operators.coffee

```diff
@@ -88,9 +88,13 @@ class Delete extends Operator
       @setTextRegister(@register, @editor.getSelectedText())
       @editor.transact =>
         for selection in @editor.getSelections()
-          selection.deleteSelectedText()
+          if @motion.isLinewise?() and selection.getText() is @editor.lineTextForBufferRow(@editor.getLastBufferRow())
```

Comment:
This doesn't work for a multi-line selection that contains the last line, but I have an idea of how to fix it.